<launch>

	<arg name="video_device"		default="/dev/video0" />

	<node name="usb_cam" pkg="usb_cam" type="usb_cam_node" output="screen" >
	    <param name="video_device" value="$(arg video_device)" />
	    <!-- <param name="image_width" value="640" /> -->
		<param name="image_width" value="640" />
	    <param name="image_height" value="480" />
	    <param name="pixel_format" value="yuyv" />
	    <param name="camera_frame_id" value="usb_cam" />
	    <param name="io_method" value="mmap"/>
  	</node>

    <node name="image_view" pkg="image_view" type="image_view" respawn="false" output="screen">
        <remap from="image" to="/usb_cam/image_raw"/>
        <param name="autosize" value="true" />
    </node>

    <!-- <arg name="marker_size"          default="10.5" /> -->
	<arg name="marker_size"          default="13.59" />
    <arg name="max_new_marker_error" default="0.08" />
    <arg name="max_track_error"      default="0.05" />

    <arg name="cam_image_topic"      default="/usb_cam/image_raw" />
    <arg name="cam_info_topic"       default="/usb_cam/camera_info" />
    <arg name="output_frame"         default="/usb_cam" />
	<!-- <arg name="output_frame"         default="/ar_marker_0" /> -->

    <node name="ar_track_alvar" pkg="ar_track_alvar" type="individualMarkersNoKinect" respawn="false" output="screen">
		<param name="marker_size"           type="double" value="$(arg marker_size)" />
		<param name="max_new_marker_error"  type="double" value="$(arg max_new_marker_error)" />
		<param name="max_track_error"       type="double" value="$(arg max_track_error)" />
		<param name="output_frame"          type="string" value="$(arg output_frame)" />

		<remap from="camera_image"  to="$(arg cam_image_topic)" />
		<remap from="camera_info"   to="$(arg cam_info_topic)" />
	</node>
	<node name="rviz" type="rviz" pkg="rviz" args="-d $(find ros_stuff)/rviz/default.rviz">
	</node>


	<arg name="robot_id"         	default="1" />
    <arg name="object_id" 		 	default="-1" />
	<arg name="base_id"          	default="3" />
    <arg name="corner_id" 	     	default="4" />

	<node name="state_publisher" pkg="ros_stuff" type="state_publisher.py" output="screen">
		<param name="robot_id"           type="int" value="$(arg robot_id)" />
		<param name="object_id"          type="int" value="$(arg object_id)" />
		<param name="base_id"       	 type="int" value="$(arg base_id)" />
		<param name="corner_id"          type="int" value="$(arg corner_id)" />
	</node>
</launch>
